### PR TITLE
feat(app): add modal for deleting protocol in ProtocolDashboard

### PR DESCRIPTION
**Pull request by koji**
# Overview
add a confirmation modal when deleting a protocol in ProtocolDashboard. Also move ProtocolDashboard related components under organisms.

# Test Plan
- Go to ProtocolDashboard
- LongPress a protocol and delete the protocol
- LongPress a protocol and pin the protocol then longPress again and delete the protocol

# Changelog
- move components under `/organisms/OnDeviceDisplay/ProtocolDashboard` to `/organisms/ProtocolDashboard`
- export ProtocolCard component
- add DeleteProtocolConfirmationModal and test

# Review requests

# Risk assessment
low

### CODE REVIEW

File: app/src/assets/localization/en/protocol_list.json [hunk 1]
```diff
@@ -15,5 +15,6 @@
   "start_setup": "Start setup",
   "this_protocol_will_be_trashed": "This protocol will be moved to this computer’s trash and may be unrecoverable.",
   "updated": "Updated",
-  "yes_delete_this_protocol": "Yes, delete protocol"
+  "yes_delete_this_protocol": "Yes, delete protocol",
+  "delete_protocol_message": " and its run history will be permanently deleted."
```
Comment (ewagoner) [inline app/src/assets/localization/en/protocol_list.json]: Can you keep these alphabetized?

File: app/src/pages/ProtocolDashboard/DeleteProtocolConfirmationModal.tsx [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+          console.error(`error deleting resources: ${e.message}`)
+        })
+    } else {
+      console.error(
+        'could not delete resources because the robot host is unknown'
+      )
+    }
+  }
+  return (
+    <Modal header={modalHeader} onOutsideClick={handleCloseModal}>
```
Comment (jerader) [inline app/src/pages/ProtocolDashboard/DeleteProtocolConfirmationModal.tsx]: do we want to allow a user to close the modal on outside click for this? i made the prop optional in this PR: https://github.com/Opentrons/opentrons/pull/12796

File: app/src/pages/ProtocolDashboard/index.tsx [hunk 3]
```diff
@@ -0,0 +1,385 @@
...
+              </Flex>
+            )}
+          </>
+        )}
+      </Flex>
+    </>
+  )
+}
+
+// export function ProtocolCard(props: {
```
Comment (jerader) [inline app/src/pages/ProtocolDashboard/index.tsx]: maybe you forgot to delete this commented out section?

**Review by jerader — APPROVED** [review]
works as expected! just left a comment and question

File: app/src/pages/ProtocolDashboard/DeleteProtocolConfirmationModal.tsx (hunk 2 above)
Comment (koji) [reply app/src/pages/ProtocolDashboard/DeleteProtocolConfirmationModal.tsx]: I'll ask about this to Mel.

File: app/src/pages/ProtocolDashboard/index.tsx (hunk 3 above)
Comment (koji) [reply app/src/pages/ProtocolDashboard/index.tsx]: thanks 😆

File: app/src/assets/localization/en/protocol_list.json (hunk 1 above)
Comment (koji) [reply app/src/assets/localization/en/protocol_list.json]: good catch, thanks!

File: app/src/pages/ProtocolDashboard/DeleteProtocolConfirmationModal.tsx (hunk 2 above)
Comment (koji) [reply app/src/pages/ProtocolDashboard/DeleteProtocolConfirmationModal.tsx]: you are right we shouldn't allow users to close the modal by outside click.
thanks!

I'll take your implementation.

**Review by ewagoner — APPROVED** [review]

**Pull request merged**